articles/tutorials/advanced/2d_shaders/07_sprite_vertex_effect/index.md

```diff
@@ -0,0 +1,863 @@
+
+
+The vertex shader declares a semantic for each input using the `:` syntax:
+
+```hlsl
+// the POSITION0 is the semantic 
+float4 position	: POSITION0,
+```
+
+>[!warning] 
+> You cannot change the `SpriteBatch` vertex shader. 
+> 
+> The `SpriteBatch` class does not offer any way to change the vertex semantics that are passed to the shader function.   
+
+### Output Semantics
+
+The same concept of semantics applies to the output of the shader. Here is the output type of the vertex shader function. Notice that the fields also have the `:` semantic syntax. These semantics instruct the graphics pipeline how to use the data:
+
+```hlsl
+struct VSOutput
+{
+	float4 position		: SV_Position;
+	float4 color		: COLOR0;
+    float2 texCoord		: TEXCOORD0;
+};
+```
+
+This is the _input_ struct for the standard pixel shaders from previous chapters. Notice how the fields are named slightly differently, but the _semantics_ are identical:
+
+```hlsl
+struct VertexShaderOutput  
+{  
+   float4 Position : SV_POSITION;  
+   float4 Color : COLOR0;  
+   float2 TextureCoordinates : TEXCOORD0;  
+};
+```
+
+### Matrix Transform
+
+The default sprite vertex shader uses this line:
+
+```hlsl
+output.position = mul(position, MatrixTransform);
+```
+
+The reason this line exists is to convert the vertices from world-space to clip-space. A vertex is a 3d coordinate in "world-space". But a monitor is a 2d display. Often, the screen's 2d coordinate system is called "clip-space". The vertex shader is converting the 3d world-space coordinate into a 2d clip-space coordinate. That conversion is a vector and matrix multiplication, using the `MatrixTransform`. 
+
+The `MatrixTransform` is computed by the [`SpriteEffect`](xref:Microsoft.Xna.Framework.Graphics.SpriteEffect) class. The full source is available, [`here`](https://github.com/MonoGame/MonoGame/blob/develop/MonoGame.Framework/Graphics/Effect/SpriteEffect.cs#L63). The relevant lines are copied below:
+
+```csharp
+// cache the shader parameter for the MatrixTransform
+_matrixParam = Parameters["MatrixTransform"];
+
+// ... some code left out for readability
+
+// create a projection matrix in the _projection variable
+Matrix.CreateOrthographicOffCenter(0, vp.Width, vp.Height, 0, 0, -1, out _projection);
+
+// ... some code left out for readability 
+
+// assign the projection matrix to the MatrixTransform
+_matrixParam.SetValue(_projection);
+```
+
+There are two common types of project matrices, 
+1. Orthographic,
+2. Perspective
+
+The orthographic projection matrix produces the classic 2d sprite effect, where sprites have no perspective when they are on the sides of the screen. 
+
+
+## Custom Vertex Shader
```

Comment:
A key fact missing in soooo many shaders!  SO great job pointing out best practice